modules/apps/company/company-test/src/testIntegration/java/com/liferay/company/service/test/CompanyLocalServiceDBPartitionTest.java

```diff
@@ -357,6 +360,58 @@ public void testAddDBPartitionCompanyWhenDBPartitionUtilFails()
 		}
 	}
 
+	@Test
+	public void testCacheCleanup() throws Exception {
+		_company1 = CompanyTestUtil.addCompany();
+
+		ClassName expectedClassName1 = _classNameLocalService.addClassName(
+			_CLEANUP_CLASS_NAME_VALUE1);
+		ClassName expectedClassName2 = _classNameLocalService.addClassName(
+			_CLEANUP_CLASS_NAME_VALUE2);
+
+		long counter = 0;
+
+		String counterName = RandomTestUtil.randomString();
+
+		try (SafeCloseable safeCloseable =
+				CompanyThreadLocal.setCompanyIdWithSafeCloseable(
+					_company1.getCompanyId())) {
+
+			_classNameLocalService.addClassName(_CLEANUP_CLASS_NAME_VALUE2);
+
+			_classNameLocalService.addClassName(_CLEANUP_CLASS_NAME_VALUE1);
+
+			counter = _counterLocalService.increment(counterName);
+
+			_counterLocalService.reset(counterName, 100000);
+		}
+
+		String virtualHostname = _company1.getVirtualHostname();
+
+		companyLocalService.deleteCompany(_company1);
+
+		_company1 = companyLocalService.copyDBPartitionCompany(
```

Comment:
@marianoalvarosaiz @achaparro The integration test logic does not make sense to me.

After I delete _company1, when I copy it? there should be nothing to copy for an removed company.

Reply:
@tinatian, we have 2 companies (TestPropsValues.getCompanyId() and _company1), we remove company1 to recreate it from TestPropsValues.getCompanyId() using the companyId from company1.

This a common use case for our customers. They have PROD and UAT, they must remove UAT from time to time to recreate it using PROD.

Please, confirm it is clear now.

Reply:
Hi @tinatian,

We have two companies (TestPropsValues.getCompanyId() and company1), we remove company1 to recreate it using TestPropsValues.getCompanyId()  but the original company1.companyId.

This is a common use case for our customers. They have PROD and UAT and they have to recreate UAT from time to time.